crates/trie/db/src/state.rs

```diff
@@ -108,13 +109,27 @@ pub trait DatabaseStateRoot<'a, TX>: Sized {
     /// The state root for this [`HashedPostState`].
     fn overlay_root(tx: &'a TX, post_state: HashedPostState) -> Result<B256, StateRootError>;
 
+    /// Calculate the state root for this [`HashedPostStateSorted`], skipping the unsorted map
+    /// construction and sort.
+    fn overlay_root_sorted(
```

Comment:
Rather than add new methods, I would prefer to modify the existing ones to accept the sorted forms of HashedPostState and TrieInput. Each of this is already essentially calling `into_sorted` immediately inside the implementation, e.g. [here](https://github.com/paradigmxyz/reth/blob/4f5b842543197dd28faea9f9424a6e46432341e3/crates/trie/db/src/state.rs#L179).